### ISSUE TITLE
(logs): improve free text search experience 

### ISSUE THREAD

**Issue by srikanthccv** [SigNoz/signoz#2859]
## Existing query builder

Users can type any text in the search bar and get the results that contain the text. While this is good for initial filtering, we observed that the expression parser on both sides fails to recognise it when adding additional filters. For instance, the user wants to search for `ImportError`. The expression will be `ImportError`; upon adding a new service filter service_name=frontend, the new expression will be `ImportError AND service_name IN ('frontend')`, which is not accepted.

Many do not read the docs. We should make this intuitive, functional and avoid confusion. We will turn the free text `ImportError` into either `body contains 'ImportError'` or `fulltext contains 'ImportError'`, which will make the detail upfront. 

## New query builder (logs specific)

We will allow the same experience in the new Where filter. In the new builder, Pressing Enter after inputting text will become a chip where `key` and `op` is fixed to `body` and `CONTAINS` with text as value.

**Comment by srikanthccv** [comment]
Notes from today's discussion: @pranay01 

We recollected the approach we discussed earlier and will implement it. When a user types a text that doesn't exist or match with any attributes, the backend returns no data—this can mean two things in logs, 1. custom key 2. full-text search. Today we show the key and all possible operators; as an example, if the text is `hello`, we show `hello = `, `hello != ` ..., to support the text search, we will show `hello` as the first option without any operator and upon selection a new chip will be made with `body CONTAINS hello`.
